crud/crud.cpp

```diff
@@ -353,20 +403,91 @@ crud::handle_has_db(const database_msg& request, std::shared_ptr<bzn::session_ba
     LOG(warning) << "session no longer available. HAS DB not executed.";
 }
 
+
+std::pair<bool, Json::Value>
+crud::get_database_permissions(const bzn::uuid_t& uuid) const
+{
+    // does the db exist?
+    if (this->storage->has(PERMISSION_UUID, uuid))
+    {
+        auto perms_data = this->storage->read(PERMISSION_UUID, uuid);
+
+        if (!perms_data)
+        {
+            throw std::runtime_error("Failed to read database permission data!");
+        }
+
+        Json::Reader reader;
+        Json::Value json;
+
+        if (!reader.parse(*perms_data, json))
+        {
+            throw std::runtime_error("Failed to parse database json permission data: " + reader.getFormattedErrorMessages());
+        }
+
+        return {true, json};
+    }
+
+    return {false, Json::Value()};
+}
+
+
+bzn::value_t
+crud::create_permission_data(const bzn::caller_id_t& caller_id) const
+{
+    Json::Value json;
+
+    json[OWNER_KEY] = caller_id;
+    json[WRITERS_KEY] = Json::Value();
+
+    LOG(debug) << "created db perms: " << json.toStyledString();
+
+    return json.toStyledString();
+}
+
+
+bool
+crud::is_caller_owner(const bzn::caller_id_t& caller_id, const Json::Value& json) const
+{
+    return json[OWNER_KEY] == caller_id;
+}
+
+
+bool
+crud::is_caller_a_writer(const bzn::caller_id_t& caller_id, const Json::Value& json) const
+{
+    for(const auto& writer_id : json[WRITERS_KEY])
+    {
+        if (writer_id == caller_id)
+        {
+            return true;
+        }
+    }
+
+    return this->is_caller_owner(caller_id, json);
```

Comment:
ah, nice! If the caller id is not in the writers list, check if the caller id is the owner!